tests/integrations/test_identity.py

```diff
@@ -0,0 +1,113 @@
+import json
+import time
+from pathlib import Path
+from typing import Generator
+
+import jwt
+import pytest
+
+from cryptography.hazmat.primitives.asymmetric import rsa
+from cryptography.hazmat.primitives import serialization
+from openeogeotrellis.integrations.identity import IDPTokenIssuer, IDP_TOKEN_ISSUER
+from openeogeotrellis.testing import gps_config_overrides
+
+
+def test_idptokenissuer_should_not_be_instantiated_manually():
+    # WHEN someone wants to manually create an instance of the IDPTokenIssuer
+    # THEN they should get an exception
+    with pytest.raises(RuntimeError):
+        IDPTokenIssuer()
+
+
+def test_without_valid_config_token_should_just_be_none(monkeypatch):
+    # GIVEN no valid config
+    with gps_config_overrides(openeo_idp_details_file=Path("/tmp/file_that_do3s_not_exist.arr")):
+        # WHEN we get a token for a job of a specific user
+        token = IDP_TOKEN_ISSUER.get_job_token("subject", "userA", "j-20250120231301301301013")
+        # THEN the token is None
+        assert token is None
+
+
+def create_idp_config(file_path: Path) -> str:
+    private_key = rsa.generate_private_key(
+        public_exponent=65537,
+        key_size=2048
+    )
+
+    pem_private_key = private_key.private_bytes(
+        encoding=serialization.Encoding.PEM,
+        format=serialization.PrivateFormat.TraditionalOpenSSL,
+        encryption_algorithm=serialization.NoEncryption(),
+    ).decode("utf-8")
+    pem_public_key = private_key.public_key().public_bytes(
+        encoding=serialization.Encoding.PEM,
+        format=serialization.PublicFormat.PKCS1
+    ).decode("utf-8")
+    with open(file_path, 'w') as fh:
+        json.dump(
+            {
+              "issuer": "https://pytests.localhost",
+              "private_key": pem_private_key,
+              "public_key": pem_public_key,
+            },
+            fh
+        )
+    return pem_public_key
+
+
+@pytest.fixture
+def idp_token_issuer(monkeypatch, tmp_path) -> Generator[str, None, None]:
```

Comment:
`monkeypatch` is not used, so do you have to declare it

Reply:
Nope, oversight of refactoring will fix